iml-agent/src/action_plugins/check_ha.rs

```diff
@@ -38,14 +48,31 @@ fn create<'a>(elem: &Element, group: impl Into<Option<&'a str>>) -> ResourceAgen
 
 fn process_resource_list(output: &[u8]) -> Result<Vec<ResourceAgentInfo>, ImlAgentError> {
     let element = Element::from_reader(output)?;
-
+    let templates: HashMap<String, ResourceAgentType> = element
+        .find_all("template")
+        .map(|elem| {
+            (
+                elem.get_attr("id").unwrap_or("").to_string(),
```

Comment:
nit: use `.unwrap_or_default()` instead:

`elem.get_attr("id").unwrap_or_default().to_string()`

Reply:
More importantly, this has the potential to use an empty `String` as a `HashMap` key.

Shouldn't we `.filter` empty keys rather than pushing an empty key into the map?